### ISSUE TITLE
Invalid JSON schema does not error out

### ISSUE THREAD

**Issue by brianshih1** [santhosh-tekuri/jsonschema#161]
Hello! I am testing out adding resource with `invalid json`, such as `{}A`. In this example, the trailing character is invalid. However, `AddResource` does not return an error.

Here is how to reproduce the error:

```
func TestInvalidJsonSchema(t *testing.T) {
	if err := jsonschema.NewCompiler().AddResource("schema.json", strings.NewReader("{}A")); err == nil {
		t.Error("error expected")
	}
}
```

I was expecting `decoder.Token)` to not return nil in [this line of code](https://github.com/santhosh-tekuri/jsonschema/blob/master/resource.go#L273). But it returns `nil`.

**Comment by santhosh-tekuri** [comment]
fixed with 620600d

**Comment by brianshih1** [comment]
thanks for the prompt fix!!
